File: frontend/src/types/transaction.ts
/**
 * Transaction status enum matching the backend C# TransactionStatus enum.
 * Represents the current state of a financial transaction.
 */
export type TransactionStatus = 'Pending' | 'Completed' | 'Failed';

/**
 * Transaction interface matching the backend C# Transaction model.
 * Represents a financial transaction in the system.
 */
export interface Transaction {
  /**
   * Unique identifier for the transaction (GUID).
   */
  transactionId: string;

  /**
   * Transaction amount (can be positive or negative, e.g., +100 or -100).
   */
  amount: number;

  /**
   * Currency code (3-letter code, e.g., "USD", "EUR").
   */
  currency: string;

  /**
   * Current status of the transaction.
   */
  status: TransactionStatus;

  /**
   * Timestamp when the transaction was created or last updated (ISO 8601 format).
   */
  timestamp: string;
}

File: backend/FinancialMonitor.Api/Models/Transaction.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace FinancialMonitor.Api.Models;

/// <summary>
/// Status of a financial transaction.
/// </summary>
public enum TransactionStatus
{
    Pending = 0,
    Completed = 1,
    Failed = 2
}

/// <summary>
/// Represents a financial transaction in the system.
/// </summary>
public class Transaction
{
    [Required]
    public Guid TransactionId { get; set; }

    [Required]
    public decimal Amount { get; set; }

    [Required]
    [StringLength(3, MinimumLength = 3, ErrorMessage = "Currency must be a 3-letter code.")]
    public string Currency { get; set; } = string.Empty;

    [Required]
    [EnumDataType(typeof(TransactionStatus), ErrorMessage = "Status must be one of the defined values: Pending, Completed, Failed.")]
    [JsonConverter(typeof(JsonStringEnumConverter))]
    public TransactionStatus Status { get; set; }

    [Required]
    public DateTime Timestamp { get; set; }
}
